File: src/seq-gcd-lcm/seq-gcd-lcm.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

long gcd(long, long);
const int range = 1000;

int main(int argc, char *argv[]) {
	srand(time(NULL));
	int i, first, second, hcf, lcm;
	const int N = atoi(argv[1]);

	for (i = 0; i <= N; i++) {
		first = rand() % range;
		second = rand() % range;

		hcf = gcd(first, second);
		lcm = (first * second)/hcf;
	}
	return 0;
}

long gcd(long first, long second) {
	if (second == 0)
		return first;
	else 
		return gcd(second, first % second);
}

File: src/seq-gcd-lcm/seq-gcd-lcm.go
package main

import (
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"time"
)

const RANGE = 1000

func main() {
	rand.Seed(time.Now().UTC().UnixNano())
	var first, second, hcf, lcm int
	N, _ := strconv.Atoi(os.Args[1])

	for i := 0; i <= N; i++ {
		first = rand.Intn(RANGE - 0)
		second = rand.Intn(RANGE - 0)

		hcf = gcd(first, second)
		lcm = (first * second) / hcf

		fmt.Printf("The greatest common divisor of %d and %d = %d \n", first, second, hcf)
		fmt.Printf("The least common multiple of %d and %d = %d \n", first, second, lcm)
	}
}

func gcd(first, second int) int {
	if second == 0 {
		return first
	} else {
		return gcd(second, first%second)
	}
}
